pkg/tenancy/tenancy.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package tenancy
+
+import (
+	"net/http"
+	"path"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// DefaultTenantHeader is the default header used to designate the tenant making a write request.
```

Comment:
As we plan to use this more generically with Query tenancy.
```suggestion
	// DefaultTenantHeader is the default header used to designate the tenant making a request.
```